nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/lazy/AbstractRenditionBuilderWork.java

```diff
@@ -136,29 +143,25 @@ void updateAndCompleteStoreEntry(List<Blob> blobs) {
         TransientStoreService tss = Framework.getService(TransientStoreService.class);
         TransientStore ts = tss.getStore(getTransientStoreName());
 
-        if (!ts.exists(key)) {
-            if (log.isDebugEnabled()) {
-                log.debug(String.format("No entry found for key %s in the %s transient store, rewriting it.", key,
-                        getTransientStoreName()));
-            }
-            RenditionService rs = Framework.getService(RenditionService.class);
-            RenditionDefinition definition = ((RenditionServiceImpl) rs).getRenditionDefinition(renditionName);
-            RenditionProvider provider = definition.getProvider();
-            if (provider instanceof AbstractLazyCachableRenditionProvider) {
-                DocumentModel doc = session.getDocument(docRef);
-                String sourceDocumentModificationDate = ((AbstractLazyCachableRenditionProvider) provider).getSourceDocumentModificationDate(
-                        doc, definition);
-                if (sourceDocumentModificationDate != null) {
-                    ts.putParameter(key, AbstractLazyCachableRenditionProvider.SOURCE_DOCUMENT_MODIFICATION_DATE_KEY,
-                            sourceDocumentModificationDate);
-                }
-            }
-
-        }
+        putSourceDocumentModificationDate(doc, ts);
         ts.putBlobs(key, blobs);
         ts.setCompleted(key, true);
     }
 
+    protected void putSourceDocumentModificationDate(DocumentModel doc, TransientStore ts) {
```

Comment:
I would have this method just extract and return the `sourceDocumentModificationDate` (or null), and let the caller deal with putting it in the `ts` if not null.